### ISSUE TITLE
Inline editing of HTMLText etc.

### ISSUE THREAD

**Issue by micschk** [sheadawson/silverstripe-blocks#27]
@sheadawson, I've been using [this module](https://github.com/milkyway-multimedia/ss-gridfield-utils) by @milkyway-multimedia for inline editing inside GridFields on a few occasions now. Thought I'd leave a quick note here; I think this would be a good addition to the Blocks module because it allows editors to edit a block's content in the context of the parent page. It supports inline HTMLEditor just fine (Uploadfield may need a little work).

![image](https://cloud.githubusercontent.com/assets/1005986/8423984/064a8572-1ef3-11e5-95d2-f18773046d11.png)

BTW; we could also look into gridfieldextensions/GridFieldAddExistingSearchButton as it would allow editors to search existing blocks by more than just title/name.

**Comment by micschk** [comment]
Only thing is, I don't think EditableRow supports versioning (yet).

**Comment by micschk** [comment]
I've been working on this a bit. @sheadawson, what do you think; better to publish as a 'block-enhancements' module or could we include this work in the Blocks module?

![image](https://cloud.githubusercontent.com/assets/1005986/13659614/7deefe70-e683-11e5-8149-530c6dbedd93.png)

**Comment by micschk** [comment]
Have everything working now: sorting to assign block area, block-preview images, inline-editing.
![image](https://cloud.githubusercontent.com/assets/1005986/13769387/69ccd7d2-ea7f-11e5-833e-24b1102f0bc3.png)

**Comment by sheadawson** [comment]
Wow this is looking pretty slick dude! How did you get on with the publishing side of things? Are there any limitations as to what fields can be used in the summary edit views? Can I have a play with it?

**Comment by micschk** [comment]
Yes, you can :) https://github.com/micschk/silverstripe-block_enhancements
